wemix/bind/backends/wemix_simulated.go

```diff
@@ -0,0 +1,270 @@
+package backends
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"math/big"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/wemix"
+	gov "github.com/ethereum/go-ethereum/wemix/bind"
+)
+
+type SimClient interface {
+	ethereum.ChainReader
+	ethereum.ChainStateReader
+	ethereum.ContractCaller
+	ethereum.GasEstimator
+	ethereum.GasPricer
+	ethereum.LogFilterer
+	ethereum.PendingStateReader
+	ethereum.PendingContractCaller
+	ethereum.TransactionReader
+	ethereum.TransactionSender
+	SuggestGasTipCap(ctx context.Context) (*big.Int, error) // GasPricer1559
+	GovContracts() *gov.GovContracts
+	Commit()
+}
+
+var ChainID = params.AllEthashProtocolChanges.ChainID
+
+type wemixSimulatedBackend struct {
+	stack      *node.Node
+	eth        *eth.Ethereum
+	backend    *backends.SimulatedBackend
+	governance *gov.GovContracts
+}
+
+func NewWemixSimulatedBackend(pk *ecdsa.PrivateKey, datadir string, alloc core.GenesisAlloc, options ...func(nodeConf *node.Config, ethConf *ethconfig.Config, envConfig *gov.InitEnvStorage)) (SimClient, error) {
```

Comment:
recommend to define and use function types

>wemix_simulated.go
type optionFn func(nodeConf *node.Config, ethConf *ethconfig.Config, envConfig *gov.InitEnvStorage)

>wemix_simulated.go
func NewWemixSimulatedBackend(pk *ecdsa.PrivateKey, datadir string, alloc core.GenesisAlloc, options ...optionFn) (SimClient, error) {

>options.go
func SetLogLevel(level int) optionFn {
}
func WithBlockGasLimit(gaslimit uint64) optionFn {
}
func WithCallGasLimit(gaslimit uint64) optionFn {
}
func WithMinerMinTip(tip *big.Int) optionFn {
}